headers-ext/src/common/cookie.rs

```diff
@@ -0,0 +1,178 @@
+
+use util::{FlatCsv, SemiColon};
+
+/// `Cookie` header, defined in [RFC6265](http://tools.ietf.org/html/rfc6265#section-5.4)
+///
+/// If the user agent does attach a Cookie header field to an HTTP
+/// request, the user agent must send the cookie-string
+/// as the value of the header field.
+///
+/// When the user agent generates an HTTP request, the user agent MUST NOT
+/// attach more than one Cookie header field.
+///
+/// # Example values
+/// * `SID=31d4d96e407aad42`
+/// * `SID=31d4d96e407aad42; lang=en-US`
+///
+#[derive(Clone, Debug, Header)]
+pub struct Cookie(FlatCsv<SemiColon>);
+
+impl Cookie {
+    pub fn get(&self, name: &str) -> Option<&str> {
```

Comment:
Could we get a `len()` method too ?

Reply:
Yea, should be easy to add, just `self.iter().count()`.